### ISSUE TITLE
M_PI not defined

### ISSUE THREAD

**Issue by micheltallon** [emmt/IPY#2]
Error in compilation on Linux Mint 20.2 (Uma).

```
ipy.c: In function Y_ipy_init:
ipy.c:439:33: error: M_PI undeclared (first use in this function)
  439 |   define_double_const("IPY_PI", M_PI);
      |                                 ^~~~
```

This is solved by adding this line in ipy.c

```
#define __USE_MISC /* For M_PI */
```

This is because in math.h M_PI is now guarded:

```
#if defined __USE_MISC || defined __USE_XOPEN
[...]
# define M_PI           3.14159265358979323846  /* pi */
[...]
#endif
```

Cheers
Michel

**Comment by micheltallon** [comment]
Actually setting __USE_XOPEN seems more general than __USE_MISC.
Indeed on Ubuntu 14.04.5, we get:

```
#if defined __USE_BSD || defined __USE_XOPEN
[...]
# define M_PI           3.14159265358979323846  /* pi */
[...]
#endif
```

**Comment by emmt** [comment]
Shoulde be solved by commit 393fbe3f1674672ac4c2146c353b0c9c6bfe9b38.
